src/Billboard/BillboardRegistry.sol

```diff
@@ -0,0 +1,241 @@
+//SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.20;
+
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "@openzeppelin/contracts/utils/Counters.sol";
+
+import "./IBillboardRegistry.sol";
+
+contract BillboardRegistry is IBillboardRegistry, ERC721 {
+    using Counters for Counters.Counter;
+
+    Counters.Counter private tokenIds;
+
+    bool public isOpened = false;
+
+    address public admin;
+
+    address public operator;
+
+    mapping(uint256 => Board) public boards;
+
+    constructor(
+        address admin_,
+        address operator_,
+        string memory name_,
+        string memory symbol_
+    ) ERC721(name_, symbol_) {
+        admin = admin_;
+        operator = operator_;
+    }
+
+    //////////////////////////////
+    /// Modifier
+    //////////////////////////////
+
+    modifier isValidAddress(address value_) {
+        if (value_ == address(0)) {
+            revert InvalidAddress();
+        }
+        _;
+    }
+
+    modifier isValidBoard(uint256 tokenId_) {
+        uint256 latestId = tokenIds.current();
+
+        if (tokenId_ < 1 || tokenId_ > latestId) {
```

Comment:
If there is L49, then this is no needed?

Reply:
Remove the first validation since ownership checker is added